bitcoin/src/blockdata/witness.rs

```diff
@@ -206,6 +188,66 @@ crate::internal_macros::define_extension_trait! {
     }
 }
 
+enum P2TrSpend<'a> {
+    Key {
+        // signature: &'a [u8],
+        annex: Option<&'a [u8]>,
+    },
+    Script {
+        leaf_script: &'a Script,
+        control_block: &'a [u8],
+        annex: Option<&'a [u8]>,
+    },
+}
+
+impl<'a> P2TrSpend<'a> {
+    /// Parses `Witness` to determine what kind of spend this is.
+    fn from_witness(witness: &'a Witness) -> Option<Self> {
+        // BIP341 says:
+        //   If there are at least two witness elements, and the first byte of
+        //   the last element is 0x50, this last element is called annex a
+        //   and is removed from the witness stack.
+        //
+        // However here we're not removing anything, so we have to adjust the numbers to account
+        // for the fact that annex is still there.
+        match witness.len() {
+            0 => None,
+            1 => Some(P2TrSpend::Key { /* signature: witness.last().expect("len > 0") ,*/ annex: None }),
+            2 if witness.last().expect("len > 0").starts_with(&[TAPROOT_ANNEX_PREFIX]) => {
+                let spend = P2TrSpend::Key {
+                    // signature: witness.second_to_last().expect("len > 1"),
```

Comment:
I'm curious why this is commented out.  Does it add meaning?

Reply:
Originally, I wrote it assuming some method might need it. There's none yet, so I got warnings. I could delete it but thought that if I went through the trouble of adding it I might keep the commented version around in case we will need it - we can just uncomment. It also serves as a reminder that key spend also has a signature and this would be its type if it was stored there.